src/Illuminate/Support/Facades/Date.php

```diff
@@ -0,0 +1,156 @@
+ * @method static \Illuminate\Support\Carbon maxValue()
+ * @method static \Illuminate\Support\Carbon minValue()
+ * @method static void mixin($mixin)
+ * @method static \Illuminate\Support\Carbon now($tz = null)
+ * @method static \Illuminate\Support\Carbon parse($time = null, $tz = null)
+ * @method static string pluralUnit(string $unit)
+ * @method static void resetMonthsOverflow()
+ * @method static void resetToStringFormat()
+ * @method static void resetYearsOverflow()
+ * @method static void serializeUsing($callback)
+ * @method static \Illuminate\Support\Carbon setHumanDiffOptions($humanDiffOptions)
+ * @method static bool setLocale($locale)
+ * @method static void setMidDayAt($hour)
+ * @method static \Illuminate\Support\Carbon setTestNow($testNow = null)
+ * @method static void setToStringFormat($format)
+ * @method static void setTranslator(\Symfony\Component\Translation\TranslatorInterface $translator)
+ * @method static \Illuminate\Support\Carbon setUtf8($utf8)
+ * @method static void setWeekEndsAt($day)
+ * @method static void setWeekStartsAt($day)
+ * @method static void setWeekendDays($days)
+ * @method static bool shouldOverflowMonths()
+ * @method static bool shouldOverflowYears()
+ * @method static string singularUnit(string $unit)
+ * @method static \Illuminate\Support\Carbon today($tz = null)
+ * @method static \Illuminate\Support\Carbon tomorrow($tz = null)
+ * @method static void useMonthsOverflow($monthsOverflow = true)
+ * @method static \Illuminate\Support\Carbon useStrictMode($strictModeEnabled = true)
+ * @method static void useYearsOverflow($yearsOverflow = true)
+ * @method static \Illuminate\Support\Carbon yesterday($tz = null)
+ */
+class Date extends Facade
+{
+    /**
+     * Get the registered name of the component.
+     *
+     * @return string
+     *
+     * @throws \RuntimeException
+     */
+    protected static function getFacadeAccessor()
+    {
+        return 'date';
+    }
+
+    /**
+     * Handle dynamic, static calls to the object.
+     *
+     * @param  string  $method
+     * @param  array   $args
+     *
+     * @return mixed
+     *
+     * @throws \RuntimeException
+     */
+    public static function __callStatic($method, $args)
+    {
+        $root = null;
+
+        try {
+            $root = static::getFacadeRoot();
+        } catch (ReflectionException $exception) {
+            // We mute reflection exceptions as if have Carbon as fallback if the facade root is missing.
+        }
+
+        if (! $root) {
+            // Use Carbon class as root if not set
+            Date::swap($root = Carbon::class);
+        }
+
+        // If Date::swap(function () { ... }) with some closure or any callable has been called
+        if (is_callable($root)) {
+            // Then we create the date with Carbon then we pass it to this callable that can modify or replace
+            // This date object with any other value
+            return $root(Carbon::$method(...$args));
```

Comment:
This feature appears to be inconsistent with Laravel facades. Calling `swap` should replace the object/class responsible for handling theat facade. This is an interceptor / adapter taking action by interpreting a special type of swap, which wasn't a real swap.
Perhaps if people want to intercept their date objects, they should wrap it in their own object / facade.
Real time facades are pretty good at that.

Reply:
@deleugpn This provide a way far easier to convert date objects rather than creating an other class with again a __callStatic method to handle all possible static creations and swap to this class. I'm not in favor of a pure deletion of this.